lib/ansible/modules/network/icx/icx_config.py

```diff
@@ -0,0 +1,483 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: icx_config
+version_added: "2.9"
+author: "Ruckus Wireless (@Commscope)"
+short_description: Manage configuration sections of Ruckus ICX 7000 series switches
+description:
+  - Ruckus ICX configurations use a simple block indent file syntax
+    for segmenting configuration into sections.  This module provides
+    an implementation for working with ICX configuration sections in
+    a deterministic way.
+notes:
+  - Tested against ICX 10.1
```

Comment:
In the notes section, you can point to the icx platform guide https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/network/user_guide/platform_icx.rst

Refer: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/junos/junos_interfaces.py#L107
Applicable for other icx modules as well.

Reply:
If you see only for junos such a note is there as it supports netconf. All other modules which support only ssh connection do not have such a note. Hence we too dint add. Hope that makes sense

Reply:
How would use come to know which connection type to use for icx modules?

Reply:
As far as other platforms modules are concerned there is a link to the platform user guide in the notes section since it supported more than one connection type, For example, refer https://docs.ansible.com/ansible/latest/modules/ios_config_module.html
If it is missed for some module it is doc related issue.